src/core/server/saved_objects/service/lib/integration_tests/repository_with_proxy_utils.ts

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import Hapi from '@hapi/hapi';
+import { IncomingMessage } from 'http';
+import { kibanaPackageJson as pkg } from '@kbn/utils';
+
+// proxy setup
+const defaultProxyOptions = (hostname: string, port: string) => ({
+  host: hostname,
+  port,
+  protocol: 'http' as 'http',
+  passThrough: true,
+});
+
+let proxyInterrupt: string | null | undefined = null;
+
+export const setProxyInterrupt = (testArg: string | null) => (proxyInterrupt = testArg);
+
+// passes the req/response directly as is
+const relayHandler = (h: Hapi.ResponseToolkit, hostname: string, port: string) => {
+  return h.proxy(defaultProxyOptions(hostname, port));
+};
+
+const proxyResponseHandler = (h: Hapi.ResponseToolkit, hostname: string, port: string) => {
+  return h.proxy({
+    ...defaultProxyOptions(hostname, port),
+    // eslint-disable-next-line @typescript-eslint/no-shadow
+    onResponse: async (err, res, request, h, settings, ttl) => proxyOnResponseHandler(res, h),
+  });
+};
+
+// mimics a 404 'unexpected' response from the proxy
+const proxyOnResponseHandler = async (res: IncomingMessage, h: Hapi.ResponseToolkit) => {
+  return h
+    .response(res)
+    .header('x-elastic-product', 'somethingitshouldnotbe', { override: true })
+    .code(404);
+};
+
+const kbnIndex = `.kibana_${pkg.version}`;
+
+// GET /.kibana_8.0.0/_doc/{type*} route (repository.get calls)
+export const declareGetRoute = (hapiServer: Hapi.Server, hostname: string, port: string) =>
+  hapiServer.route({
+    method: 'GET',
+    path: `/${kbnIndex}/_doc/{type*}`,
+    options: {
+      handler: (req, h) => {
+        if (req.params.type === 'my_type:myTypeId1' || req.params.type === 'my_type:myType_123') {
```

Comment:
Looking at the complexity of the setup, I'm wondering what if we have just a single smoke test for this scenario instead of covering all the methods?